crates/executor/src/execution_state.rs

```diff
@@ -1,14 +1,24 @@
 use std::sync::Arc;
 
+use crate::state_reader::LruCachedReader;
 use blockifier::{
     block_context::BlockContext,
     state::cached_state::{CachedState, GlobalContractCache},
 };
+use cached::SizedCache;
 use pathfinder_common::{BlockNumber, BlockTimestamp, ChainId, SequencerAddress, StateUpdate};
 use primitive_types::U256;
+use std::sync::Mutex;
 
 use super::state_reader::PathfinderStateReader;
 
+lazy_static::lazy_static!(
+    static ref CONTRACT_CACHE: GlobalContractCache = {
+        let inner = SizedCache::with_size(16 * 1024 * 1024);
```

Comment:
Isn't 16M the max number of items stored in the cache? If yes, that's probably way too high.

Reply:
It actually is :sweat_smile: I'm not sure where I got the idea it was a size in bytes.

Reply:
Fixed, TY!